### PR TITLE
Add TypeScript definition

### PR DESCRIPTION
**Pull request by Richienb**

### CODE REVIEW

**Comment by Richienb** [comment]
CI is throwing these errors:

```
  index.d.ts:1:26
  ✖  1:26  Cannot find module node:tty or its corresponding type declarations.      

  index.test-d.ts:1:29
  ✖  1:29  Cannot find module node:process or its corresponding type declarations.  
```

**Comment by Richienb** [comment]
Unsure whether `@types/node` should be a dependency.

File: index.d.ts [hunk 1]
```diff
@@ -0,0 +1,55 @@
+import {WriteStream} from 'node:tty';
+
+export interface Options {
+	/**
+	Whether `process.argv` should be sniffed for --color and --no-color flags.
+
+	@default true
+	*/
+	sniffFlags?: boolean;
```
Comment (sindresorhus) [inline index.d.ts:9]:
```suggestion
	readonly sniffFlags?: boolean;
```

File: index.d.ts [hunk 2]
```diff
@@ -0,0 +1,55 @@
+import {WriteStream} from 'node:tty';
+
+export interface Options {
+	/**
+	Whether `process.argv` should be sniffed for --color and --no-color flags.
```
Comment (sindresorhus) [inline index.d.ts:5]:
```suggestion
	Whether `process.argv` should be sniffed for `--color` and `--no-color` flags.
```

**Comment by sindresorhus** [comment]
> Unsure whether @types/node should be a dependency.

Dev

**Pull request merged**